Modules/readline.c

```diff
@@ -379,90 +421,114 @@ set_completion_display_matches_hook(PyObject *self, PyObject *args)
+    return set_hook("startup_hook", &readlinestate_global->startup_hook,
+            function);
 }
 
-PyDoc_STRVAR(doc_set_startup_hook,
-"set_startup_hook([function]) -> None\n\
-Set or remove the function invoked by the rl_startup_hook callback.\n\
-The function is called with no arguments just\n\
-before readline prints the first prompt.");
-
-
 #ifdef HAVE_RL_PRE_INPUT_HOOK
 
 /* Set pre-input hook */
 
+/*[clinic input]
+readline.set_pre_input_hook
+
+    function: object = None
+    /
+
+Set or remove the function invoked by the rl_pre_input_hook callback.
+
+The function is called with no arguments after the first prompt
+has been printed and just before readline starts reading input
+characters.
+[clinic start generated code]*/
+
 static PyObject *
-set_pre_input_hook(PyObject *self, PyObject *args)
+readline_set_pre_input_hook_impl(PyObject *module, PyObject *function)
+/*[clinic end generated code: output=fe1a96505096f464 input=4f3eaeaf7ce1fdbe]*/
 {
-    return set_hook("pre_input_hook", &readlinestate_global->pre_input_hook, args);
+    return set_hook("pre_input_hook", &readlinestate_global->pre_input_hook,
+            function);
 }
-
-PyDoc_STRVAR(doc_set_pre_input_hook,
-"set_pre_input_hook([function]) -> None\n\
-Set or remove the function invoked by the rl_pre_input_hook callback.\n\
-The function is called with no arguments after the first prompt\n\
-has been printed and just before readline starts reading input\n\
-characters.");
-
 #endif
 
 
 /* Get the completion type for the scope of the tab-completion */
+
+/*[clinic input]
+readline.get_completion_type
+
+Get the type of completion being attempted.
+[clinic start generated code]*/
+
 static PyObject *
-get_completion_type(PyObject *self, PyObject *noarg)
+readline_get_completion_type_impl(PyObject *module)
+/*[clinic end generated code: output=5c54d58a04997c07 input=04b92bc7a82dac91]*/
 {
   return PyLong_FromLong(rl_completion_type);
 }
 
-PyDoc_STRVAR(doc_get_completion_type,
-"get_completion_type() -> int\n\
-Get the type of completion being attempted.");
+/* Get the beginning index for the scope of the tab-completion */
 
+/*[clinic input]
+readline.get_begidx
 
-/* Get the beginning index for the scope of the tab-completion */
+Get the beginning index of the completion scope.
+[clinic start generated code]*/
 
 static PyObject *
-get_begidx(PyObject *self, PyObject *noarg)
+readline_get_begidx_impl(PyObject *module)
+/*[clinic end generated code: output=362616ee8ed1b2b1 input=e083b81c8eb4bac3]*/
 {
     Py_INCREF(readlinestate_global->begidx);
     return readlinestate_global->begidx;
 }
 
-PyDoc_STRVAR(doc_get_begidx,
-"get_begidx() -> int\n\
-get the beginning index of the completion scope");
+/* Get the ending index for the scope of the tab-completion */
 
+/*[clinic input]
+readline.get_endidx
 
-/* Get the ending index for the scope of the tab-completion */
+get the ending index of the completion scope
```

Comment:
Please fix also other docstrings.
```suggestion
Get the ending index of the completion scope.
```